docs/http/http-spans.md

```diff
@@ -295,52 +298,58 @@ See the examples for more details about:
-[`WSGIScriptAlias` directive][modwsgisetup] of `mod_wsgi`).
+HTTP server frameworks and their instrumentations have a limited knowledge about the HTTP infrastructure and intermediaries request went through. In a general case, they can only use HTTP request properties such as request target or headers to populate `server.*` attributes.
 
-An application can be "mounted" under an **application root**
-(also known as a *[context root][]*, *[context prefix][]*, or *[document base][]*)
-which is a fixed path prefix of the URL that determines to which application a request is routed
-(e.g., the server could be configured to route all requests that go to an URL path starting with `/webshop/`
-at a particular virtual host
-to the `com.example.webshop` web application).
+Certain web frameworks may provide additional information about server. For example CGI specification defines `SERVER_NAME` and `SERVER_PORT` ([RFC 3875][rfc-servername]) variables. HTTP servers usually know server names as they are provided by users in the configuration. See for example, the Apache [`ServerName`][ap-sn] or NGINX [`server_name`][nx-sn] directive.
 
-Some servers allow to bind the same HTTP application to multiple `(virtual host, application root)` pairs.
+Still, the original logical server name might not be available to the HTTP server instrumentation or be inaccurate.
 
-> TODO: Find way to trace HTTP application and application root ([opentelemetry/opentelementry-specification#335][])
+#### Setting `server.address` and `server.port` attributes
 
-[HTTP host header]: https://tools.ietf.org/html/rfc7230#section-5.4
-[PEP 3333]: https://www.python.org/dev/peps/pep-3333/
-[modwsgisetup]: https://modwsgi.readthedocs.io/en/develop/user-guides/quick-configuration-guide.html
-[context root]: https://docs.jboss.org/jbossas/guides/webguide/r2/en/html/ch06.html
-[context prefix]: https://marc.info/?l=apache-cvs&m=130928191414740
-[document base]: http://tomcat.apache.org/tomcat-5.5-doc/config/context.html
+In the context of HTTP server, `server.address` and `server.port` attributes capture logical host and port name and intended, whenever possible, to be the same on the client and server sides.
+
+HTTP server instrumentations SHOULD do the best effort when populating `server.address` and `server.port` attributes and determine them by using the first of the following that applies:
+
+* The original host which may be passed by the reverse proxy in the `host` identifier of the [`Forwarded`][Forwarded] header or in the [`X-Forwarded-Host`][X-Forwarded-Host] header.
+* Server name and port available through web framework or HTTP server properties if they're based on the user configuration. Instrumentations SHOULD NOT use properties that rely on the `Host` header.
+* The [request target][request target] if it's passed in the absolute form.
+* The `Host` header.
```

Comment:
Fallback for HTTP/1.0 where no `Host` header? (`network.local.address/port`?)

Reply:
Let's figure out `server.address` requirement level first (#424).
If it stays recommended, then I'd prefer to not provide a fallback option.

HTTP 1.0 is [obsolete since 1997](https://en.wikipedia.org/wiki/HTTP#:~:text=Summary%20of%20HTTP%20milestone%20versions) and it would be nice not to add features for it specifically.

Reply:
> Let's figure out `server.address` requirement level first (#424). If it stays recommended, then I'd prefer to not provide a fallback option.

I'm good with keeping `server.address` as (only) recommended, and sent #429 with your proposal for fixing the requirement level on `server.port` instead.

I'm not sure (yet) about not having a fallback.

We've discussed the idea that reconstructing the original URL on the server side is an important use case.

I think users seeing something like `http://123.123.123.123:8080/petclinic` is potentially more useful than not seeing anything.

Reply:
If address is recommended, then backends should expect that it might be missing. By keeping it recommended, we're also assuming that URL reconstruction is not essential. But it can still be done by falling back (on the consumer) to network attributes.

It also feels like something that can be changed post-stability without breaking based on real-world feedback.